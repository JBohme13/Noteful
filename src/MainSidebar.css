#folder {
    height: 80px;
    text-align: center;
    margin-top: 2.5px;
    margin-bottom: 2.5px;
    border: 1px grey solid;
    display: flex;
    flex-direction: column;
}

#folder-link {
    margin-top: auto;
    margin-bottom: auto;
    text-decoration: none;
    color: darkslategrey;
    font-size: 24px;
}

#add-folder {
    border: 1px solid grey;
    border-radius: 3px;
    box-shadow: 1px 1px;
    margin: 5px;
    height: 22px;
    font-size: 14px;
    color: darkslategrey;
}

#add-folder:active {
    box-shadow: -1px -1px;
}

#delete-folder {
    border: 1px solid grey;
    border-radius: 3px;
    box-shadow: 1px 1px;
    margin: 5px;
    height: 22px;
    font-size: 14px;
    color: darkslategrey;
}

#delete-folder:active {
    box-shadow: -1px -1px;
}