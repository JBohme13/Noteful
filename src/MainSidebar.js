import React, { Component } from 'react'
import {Link} from 'react-router-dom'
import NotefulContext from './NotefulContext'
import FolderError from './FolderError'
import './MainSidebar.css'

export default class MainSidebar extends Component {
    static contextType = NotefulContext;
    render() {
        const value = this.context;
        const folders = value.folders.map((folder, i) => 
            <section key={i} className='folder'>
                <Link 
                  className='folder-link' 
                  to={`/folders/${folder.id}`}
                  onClick={() => value.setFolderId(folder.id)}
                >
                  {folder.name}
                </Link>
            </section>
        )
        return(
            <section className='sidebar-container'>
                <FolderError>
                { folders }
                </FolderError>
                <button 
                  id='add-folder'
                  onClick={() => value.history.push('/add-folder')}>
                      Add Folder
                </button>
                <button 
                  id='delete-folder'
                  onClick={() => value.history.push('/delete-folder')}>
                      Delete Folder
                </button>
            </section>
            
        )
    }
}