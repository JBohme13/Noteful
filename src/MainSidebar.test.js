import React from 'react';
import ReactDOM from 'react-dom';
import MainSidebar from './MainSidebar';
import renderer from 'react-test-renderer'

describe('MainSidebar component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MainSidebar />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<MainSidebar />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});