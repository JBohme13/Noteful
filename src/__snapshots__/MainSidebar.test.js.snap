// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSidebar component renders the UI as expected 1`] = `
<section
  className="sidebar-container"
>
  <button
    id="add-folder"
    onClick={[Function]}
  >
    Add Folder
  </button>
  <button
    id="delete-folder"
    onClick={[Function]}
  >
    Delete Folder
  </button>
</section>
`;
